### PR TITLE
feat: Create OakVideoTranscript and related components

### PR DESCRIPTION
**Pull request by weronika-szalas**
# How to review this PR

_Leave this text block for the reviewer_

- Check [component hierarchy](https://miro.com/app/board/uXjVNnKBgyk=/?share_link_id=59445593794) is followed correctly
- Check the design [Heuristics](https://components.thenational.academy/?path=/docs/docs-howtodesigncomponents--docs#heuristics-for-component-design) have been followed
- Check [naming conventions](https://components.thenational.academy/?path=/docs/docs-namingconventions--docs) have been applied
- Check for these gotchyas:
  - Missing exports for Oak components
  - Accidental export of Internal components
  - Snapshots of unexpected components have been modified
  - Circular dependencies
  - Code duplication (via not using base components)
  - Non-functional storybook for this or related components
  - Sensitve files changed eg. atoms, or style tokens
  - Relative imports
  - Default exports

# Add your PR description below

Added the following components:
- `OakVideoTranscript`
- `OakCopyLink`
- `OakSignLanguage`

## Link to the design doc

https://www.figma.com/design/iTimVTBZ3XcszhRa9KtD7r/Cycle-2-requirements---Teachers?node-id=0-1&node-type=canvas&m=dev

## A link to the component in the deployment preview

`OakVideoTranscript`
https://deploy-preview-330--lively-meringue-8ebd43.netlify.app/?path=/docs/components-organisms-shared-oakvideotranscript--docs

`OakSignLanguageButton`
https://deploy-preview-330--lively-meringue-8ebd43.netlify.app/?path=/docs/components-molecules-oaksignlanguagebutton--docs

`OakCopyLinkButton`
https://deploy-preview-330--lively-meringue-8ebd43.netlify.app/?path=/docs/components-molecules-oakcopylinkbutton--docs

### CODE REVIEW

**Comment by RBrownsett** [comment]
@weronika-szalas looking good. Just a couple of observations:
-The copy link button has an image with the alt text 'copy' which is very similar to the actual link text so it is being marked up as redundant. We could either expand on the alt text, set it to hidden, or remove it all together? Maybe best to have it as `aria-hidden=true` ?  
image

How it has been handled on the Share to pupil page:
image

-The other slight issue (or it could be story book) is the alignment in mobile view looks like it needs padding to the right of the sign language button? Or will we be able to amend this once its in OWA?

image

**Review by Monika-Koziol — APPROVED** [review]
Question about the state of the button that says linked copied: could it be visible for longer, e.g. until a user clicks on something else?

**Comment by weronika-szalas** [comment]
> Question about the state of the button that says linked copied: could it be visible for longer, e.g. until a user clicks on something else?

We can set the timer to longer but it would be hard to implement it to be visible until the next user click.

**Comment by RBrownsett** [comment]
> > Question about the state of the button that says linked copied: could it be visible for longer, e.g. until a user clicks on something else?
> 
> We can set the timer to longer but it would be hard to implement it to be visible until the next user click.

I think it would be good for it to be a few seconds longer

**Comment by weronika-szalas** [comment]
@RBrownsett @Monika-Koziol I increased the timer to 4 seconds and buttons alignment on mobile. I will check if possible to hide alt text of the icon easily in this case as we are using a different component to build it then in Pupils.

File: src/components/molecules/OakCopyLinkButton/OakCopyLinkButton.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      const copyMessage = "Link copied";
+      setLabel(copyMessage);
+      setAnnounce(copyMessage);
+      setActive(false);
+
+      setTimeout(() => {
+        setAnnounce(""); // used for aria-live announcement
+        setActive(true);
+        setLabel("Copy link");
+      }, 4000);
```
Comment (k-huggs) [inline src/components/molecules/OakCopyLinkButton/OakCopyLinkButton.tsx:35]: why this amount of time?

Comment (weronika-szalas) [reply src/components/molecules/OakCopyLinkButton/OakCopyLinkButton.tsx:35]: a request from @Monika-Koziol and @RBrownsett in the comment above, let me put it in the const so it is more clear what is this timer for

**Review by RBrownsett — APPROVED** [review]
Approved - mobile view renders correctly and screen reader does not read out the redundant text

**Pull request merged**